include/EASTL/span.h

```diff
@@ -41,9 +43,18 @@ namespace eastl
 
 		template <typename T>
 		struct HasSizeAndData<T, void_t<decltype(eastl::size(eastl::declval<T>())), decltype(eastl::data(eastl::declval<T>()))>> : eastl::true_type {};
-	}
 
-	static EA_CONSTEXPR size_t dynamic_extent = size_t(-1);
+		// SubspanExtent
+		//
+		// Integral constant that calculates the resulting extent of a templated subspan operation.
+		//
+		//   If Count is not dynamic_extent then SubspanExtent::value is Count,
+		//   otherwise, if Extent is not dynamic_extent, SubspanExtent::value is (Extent - Offset),
+		//   otherwise, SubspanExtent::value is dynamic_extent.
+		//
+		template<size_t Extent, size_t Offset, size_t Count>
+		struct SubspanExtent : eastl::integral_constant<size_t, (Count != dynamic_extent ? Count : (Extent != dynamic_extent ? (Extent - Offset) : dynamic_extent))> {};
```

Comment:
Wasn't too sure on how to best deal with the rather lengthy subspan extent expression.

I didn't like duplicating it across its two usage locations, so I put it into the Internal namespace as an integral_constant as it felt like a safe option in terms of compiler support.

If you'd rather just be inlined into its two usage locations (or handled another way) then I'll happily accommodate.